community/kernel/src/main/java/org/neo4j/kernel/impl/api/store/StoreStatement.java

```diff
@@ -159,18 +162,33 @@ public void acquire()
     }
 
     @Override
-    public void close()
+    public void release()
+    {
+        assert acquired;
+        doRelease();
+        acquired = false;
+    }
+
+    private void doRelease()
     {
-        assert !closed;
         if ( indexReaderFactory != null )
         {
             indexReaderFactory.close();
+            // we can actually keep this object around
```

Comment:
Implementation would be `IndexReaderFactory.Caching`. It's `#close()` closes all cached index readers and clears map used to keep store them.